src/extractor/edge_based_graph_factory.cpp

```diff
@@ -431,6 +431,33 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                                                 sizeof(target_node.weight));
                         previous = target_node.node_id;
                     }
+
+                    // We also now write out the mapping between the edge-expanded edges and the
+                    // original nodes. Since each edge represents a possible maneuver, external
+                    // programs can use this to quickly perform updates to edge weights in order
+                    // to penalize certain turns.
+
+                    const bool isCompressed =
+                        m_compressed_edge_container.ContainsCompressedGeometries(edge_from_u);
+
+                    const auto &from_node =
+                        isCompressed
+                            ? m_node_info_list[m_compressed_edge_container.GetLastEdgeSourceID(
+                                  edge_from_u)]
+                            : m_node_info_list[node_u];
```

Comment:
The behaviour here looks right, but I think the wording could be changed.  I think if I were to come back to this in a few months, `isCompressed` and `ContainsCompressedGeometries` might confuse me, as technically, all edges are in the `CompressedGeometryContainer`.

How about `isTrivial` and `isTrivial()`, and the some comments in the `CompressdGeometryContainer` to explain what `Trivial` means in this context.